### DISCUSSION TITLE
brew   fails now in fresh vm.

### DISCUSSION THREAD

**Discussion by davea42** [Homebrew/discussions#4615] [Everyday usage] [answered]
### Output of `brew config`

```shell
Run brew config
HOMEBREW_VERSION: 4.0.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 54c8876dc39047c04de15e7d212979ae8d98cf1c
Last commit: 13 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ecd1fac6f6cbd13d825ef9fce65ca308c383d467
Core tap last commit: 9 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS: 3650
HOMEBREW_COLOR: set
HOMEBREW_MAKE_JOBS: 3
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 3-core 64-bit ivybridge
Clang: 14.0.0 build 1400
Git: 2.41.0 => /usr/local/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 12.6.7-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2 => /Applications/Xcode_14.2.app/Contents/Developer
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/etc/bash_completion.d/prlctl
  /usr/local/etc/bash_completion.d/prlsrvctl
  /usr/local/share/man/man8/prl_convert.8
  /usr/local/share/man/man8/prl_disk_tool.8
  /usr/local/share/man/man8/prlctl.8
  /usr/local/share/man/man8/prlexec.8
  /usr/local/share/man/man8/prlsrvctl.8
Error: Process completed with exit code 1.
```

### Output of `brew doctor`

```shell
I think this is from brew doctor... but the github output
is unclear on that point.
Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/etc/bash_completion.d/prlctl
  /usr/local/etc/bash_completion.d/prlsrvctl
  /usr/local/share/man/man8/prl_convert.8
  /usr/local/share/man/man8/prl_disk_tool.8
  /usr/local/share/man/man8/prlctl.8
  /usr/local/share/man/man8/prlexec.8
  /usr/local/share/man/man8/prlsrvctl.8
Error: Process completed with exit code 1.
```

### Description of issue

Witrhout brew config brew doctor:
```
HEAD is now at 9a482dea2 Merge pull request #15612 from Homebrew/sponsors-maintainers-man-completions
remote: fatal: object b071594362655c488a53baadd3a4048a00dd6cfe cannot be read        
remote: aborting due to possible repository corruption on the remote side.
```
With bre config brew doctor
The brew config  brew doctor  seems to find errors before
running the bash command, and exits.

I saw your discussion suggesting 
unset HOMEBREW_NO_INSTALL_FROM_API
but am uncertain where to put it.

Was working a few weeks ago.

```
jobs:
 macos_autotools:
    runs-on: macos-latest
    steps:
    - uses: actions/checkout@v2
    - name: tools
      run:  |  
        brew config
        brew doctor
        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
        brew install autoconf automake libtool
        brew install --overwrite pkgconf
    - name: autoreconf
      run: autoreconf --warnings=all  -vif
    - name: config
      run: ./configure --enable-dwarfexample --enable-wall
    - name: make
      run:  |
        make 
        make check
```

**Comment by davea42** [comment]
Added brew cleanup:
Run brew cleanup
Removing: /Users/runner/Library/Caches/Homebrew/Cask/virtualbox--6.1.38,153438.dmg... (121.9MB)
Pruned 7 symbolic links and 35 directories from /usr/local
==> This operation has freed approximately 121.9MB of disk space.
Warning: Running in non-interactive mode because `$CI` is set.
==> Checking for `sudo` access (which may request your password)...
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> HOMEBREW_NO_INSTALL_FROM_API is set.

but the bash/curl fails with
HEAD is now at f40b0f9f6 Merge pull request #15619 from Bo98/artifact-test-fix
remote: fatal: object 2bd0b6b5a735e8127d0c92011d23ac81008d3400 cannot be read        
remote: aborting due to possible repository corruption on the remote side.

(still).

**Comment by davea42** [comment]
Commenting out 2 macos builds (autotools, cmake) and leaving in macos meson build
see  macos meson failing, when it did not running all three in sequence (1 of three
would fail). Very odd.  Only macos builds fail, and apparently not entirely consistently.

**Comment by Bo98** [comment, marked as answer]
Is there a reason you are running the installer when `macos-latest` already has Homebrew installed?

**Reply by davea42** [reply]
I had no idea homebrew was installed automatically. Will remove the install and test.

**Reply by davea42** [reply]
Now working again. Thanks.
